src/databricks/sql/auth/thrift_http_client.py

```diff
@@ -0,0 +1,48 @@
+# Databricks CLI
+# Copyright 2022 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+import thrift.transport.THttpClient
+
+logger = logging.getLogger(__name__)
+
+
+class THttpClient(thrift.transport.THttpClient.THttpClient):
+
+    def __init__(self, authenticator, uri_or_host, port=None, path=None, cafile=None, cert_file=None, key_file=None, ssl_context=None):
+        super().__init__(uri_or_host, port, path, cafile, cert_file, key_file, ssl_context)
+        self.__authenticator = authenticator
+
+    def setCustomHeaders(self, headers):
+        self._headers = headers
+        super().setCustomHeaders(headers)
+
+    def flush(self):
+        # TODO retry behaviour
+        # TODO locking for multiple concurrent refresh token
+        req = {}
+        self.__authenticator.add_auth_token(req)
+        self._headers['Authorization'] = req['Authorization']
```

Comment:
Implied dependency of the function setting the `Authorization` header.

This could be more robust by naming the authenticator function `add_headers` and merging the entire return value into the `self._headers` dict. That would also allow for adding multiple headers out of the box.

Reply:
thanks for the comment. addressed.